services/autorust/codegen/src/codegen.rs

```diff
@@ -166,71 +168,102 @@ impl TypeNameCode {
         Ok(type_name_code)
     }
 
+    pub fn set_is_ref(&mut self, is_ref: bool) {
+        self.is_ref = is_ref;
+    }
```

Comment:
What are your thoughts on using the builder-style pattern here?

```suggestion
    pub fn set_is_ref(mut self, is_ref: bool) -> Self {
        self.is_ref = is_ref;
        self
    }
```

This mirrors other functions used in the impl of this struct, such as this (which is a few lines below):
```rust
    pub fn optional(mut self, optional: bool) -> Self {
        self.optional = optional;
        self
    }
```

Reply:
For this type, I'd like to move away from the builder-style pattern. I don't think it is needed. Mixing the two is not great, so I'll proceed with removing it if that works.

Reply:
I don't have strong feelings either way.  I probably should have labeled these "questions/nit".